toml.abnf

```diff
@@ -117,14 +117,19 @@ underscore = %x5F                  ; _
 
 ;; Float
 
-float = integer ( frac / ( frac exp ) / exp )
+float = integer ( exp / frac [ exp ] )
+float =/ special-float
 
 frac = decimal-point zero-prefixable-int
 decimal-point = %x2E               ; .
 zero-prefixable-int = DIGIT *( DIGIT / underscore DIGIT )
 
 exp = "e" integer
 
+special-float = ( [ minus / plus ] inf ) / nan
```

Comment:
I wonder if allowing `±` in front of `nan` also would be simpler – keeping the universal rule that numbers can have a sign prefix. Of course `+nan`, `nan` and `-nan` would all mean the same thing, but it could be simpler. Not a big issue, just a thought.

Reply:
That's an interesting point. It would keep the consistency across all values, which would be a nice behavior. I suppose if the implementation wanted to, it could carry over the sign to the NaN value, even though it's insignificant, but who knows!